pkg/util/common/common_test.go

```diff
@@ -93,26 +93,36 @@ func TestStructToMap(t *testing.T) {
 		},
 	}
 
-	assert.Equal(t, map[string]interface{}{
-		"name":  "top",
-		"value": 0,
-		"nested": map[string]interface{}{
-			"moreNested": []interface{}{
-				map[string]interface{}{
-					"id":          "toto",
-					"name":        "ms1",
-					"value":       float64(1),
-					"JSONLessStr": "42",
-				},
-				map[string]interface{}{
-					"name":        "ms2",
-					"value":       float64(2),
-					"JSONLessStr": "",
-				},
-			},
-		},
-		"mymap": map[string]interface{}{
-			"n1": map[string]interface{}{
+	topSliceOfStructs := []Top{
+		top,
+		top,
+	}
+
+	topPointer := &top
+
+	type emptyInterval interface {
+		String() string
+	}
+
+	var nilCase emptyInterval
+
+	var nilPointerCase *Top = nil
+
+	numCase := 1
+	letterCase := "a"
+
+	cases := []struct {
+		name string
+		in   interface{}
+		out  map[string]interface{}
+	}{
+		{"nil interface ", nilCase, map[string]interface{}(nil)},
```

Comment:
I would inline inputs in the table here as well, otherwise it's hard to follow. Also - it's better to specify field names in struct initialization, that way if the order in the struct changes it's not an issue.

Reply:
✔️